docs/query-builder.md

```diff
@@ -448,64 +448,161 @@ DB::collection('items')
 
 **Push**
 
-Add items to an array.
+Add one or multiple values to the items array.
 
 ```php
+// Push the value to the matched documents
 DB::collection('users')
-    ->where('name', 'John')
-    ->push('items', 'boots');
+  ->where('name', 'John')
+  // Push a single value to the items array
+  ->push('items', 'boots');
+// Result:
+// items: ['boots']
+
+DB::collection('users')
+  ->where('name', 'John')
+  // Push multiple values to the items array
+  ->push('items', ['hat', 'jeans']);
+// Result:
+// items: ['boots', 'hat', 'jeans']
 
+// Or
+
+// Push the value to an exact document
 $user->push('items', 'boots');
+$user->push('items', ['hat', 'jeans']);
 ```
 
+To add an array as a value to the messages array. value arrays must be nested into a list array, otherwise they are
+interpreted as a list of scalar values.
+
 ```php
 DB::collection('users')
-    ->where('name', 'John')
-    ->push('messages', [
-        'from' => 'Jane Doe',
-        'message' => 'Hi John',
-    ]);
+  ->where('name', 'John')
+    // Push an array as a value to the items array
+  ->push('messages', [
+      [
+          'from'    => 'Jane Doe',
+          'message' => 'Hi John',
+      ],
+  ]);
+// Result:
+// messages: [
+//      { from: "Jane Doe", message: "Hi John" }
+//  ]
+
+// Or
 
 $user->push('messages', [
-    'from' => 'Jane Doe',
-    'message' => 'Hi John',
+    [
+        'from'    => 'Jane Doe',
+        'message' => 'Hi John',
+    ],
 ]);
 ```
 
-If you **DON'T** want duplicate items, set the third parameter to `true`:
+If you **DON'T** want duplicate values, set the third parameter to `true`:
```

Comment:
I haven't checked the code, but this suggests that the boolean causes [`$addToSet`](https://www.mongodb.com/docs/manual/reference/operator/update/addToSet/) to be used internally instead of [`$push`](https://www.mongodb.com/docs/manual/reference/operator/update/push/). Is that correct?

Having a boolean argument control that rubs me the wrong way (I'll add it to the growing list), but I assume there's an explanation for this behavior. Is the `push()` method here actually derived from something else in Eloquent, or was it originally created in this library specifically for the MQL operators?

Reply:
> I haven't checked the code, but this suggests that the boolean causes [`$addToSet`](https://www.mongodb.com/docs/manual/reference/operator/update/addToSet/) to be used internally instead of [`$push`](https://www.mongodb.com/docs/manual/reference/operator/update/push/). Is that correct?

Yes. It's the logged queries of pushing normally and pushing unique values.

```plain
array:2 [
  0 => array:3 [
    "query" => "users.updateMany({"_id":"65672f58551357588a07c993"},{"$push":{"items":{"$each":["boots"]}}},{"multiple":true})"
    "bindings" => []
    "time" => 1.82
  ]
  1 => array:3 [
    "query" => "users.updateMany({"_id":"65672f58551357588a07c993"},{"$addToSet":{"items":{"$each":["hat","jeans"]}}},{"multiple":true})"
    "bindings" => []
    "time" => 0.6
  ]
]
```
 

> Having a boolean argument control that rubs me the wrong way (I'll add it to the growing list), but I assume there's an explanation for this behavior. Is the `push()` method here actually derived from something else in Eloquent, or was it originally created in this library specifically for the MQL operators?

 The `push` method is originally from Eloquent and it's used for saving a model instance and its relationships to the database. this library overrides the `push` method and accepts three args. the original method doesn't accept any argument. If no arg is passed, the original `push` method will be called.

Reply:
Is Eloquent's `push` method documented anywhere beyond [Recursively Saving Models & Relationships](https://laravel.com/docs/10.x/eloquent-relationships#the-push-method)? That doesn't appear to take _any_ arguments and the method seems entirely different in purpose (this doesn't have to do with relationships).

@GromNaN: A boolean flag to determine whether `$push` or `$addToSet` is used seems like a poor design as it makes the code less self-documenting. It may be worth making a ticket to revisit this down the line. Perhaps introduce a new `addToSet()` or `pushUnique` method and then deprecate the original.

Reply:
💯 agree that the double behavior of this method is a bad design.

If deprecate the custom behavior of `Model::push` with parameters, we have to do deprecate the `pull` method too.

Let's improve the documentation for now.